### PR TITLE
#05 : Custom Blocks.

### LINKED ISSUE #5: #05 ­- Block System
## Objective
At the end of this course, you will be able to

Create "Block Types" with fields
Place Multiple Instances of the block type with different field values
Programmatically update block instance values / content
## Exercise
- Create a Block type called "Stock Exchange Rate Card"
  - Company Name
  - Symbol
  - Last Price
  - Change
- Create Instances of the block with different sets of values and place them at different spots on the site. Example values ­ (Apple, AAPL), (Bank of America,BAC), (Transocean, RIG), (Freeport, FCX). (Ignore the Last Price and Change fields’ values, we will be dealing them in next step).
- Below API can be used to retrieve Last Price and Change values http://dev.markitondemand.com/MODApis/Api/v2/Quote/json?symbol=BAC&callback=myFunction
- Build a custom module, which on Cron run,
  - Iterates through each instance of blocks of type "Stock Exchange Price Card"
  - For each block, take the symbol value and call the API to retrieve the Last Price and Change values
  - Update the block field values programmatically with the values of Last Price and Change retrieved from the API and save the block

### PR DESCRIPTION
**Pull request by amansrivastava**
Provided API wasn't working any more.
I have used a different API to fetch Forex rates.
Updated exercise:
- Create Custom Block Type `Forex Card`.
- Add Fields `base_currency`, `target_currency` (multiselect), forex_rates
- Add config form to save Endpoint and Token.
- Create Drush Command to update `forex_rates` field based on base and target currency.
- forex_rates is not editable and can only be updated by command.

API reference: https://finnhub.io/docs/api#forex-rates

### CODE REVIEW

File: config/default/core.extension.yml [hunk 1]
```diff
@@ -1,4 +1,6 @@
 module:
+  admin_toolbar: 0
+  admin_toolbar_tools: 0
```
Comment (joshirohit100) [inline config/default/core.extension.yml:3]: unwanted changes

File: config/default/forex_rates.settings.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+api_token: bub7hsv48v6pmjua2b0g
+api_url: 'https://finnhub.io/api/v1/forex/rates'
```
Comment (joshirohit100) [inline config/default/forex_rates.settings.yml]: we shouldn't commit token or something in repo.

File: docroot/modules/custom/forex_rates/composer.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    "name": "custom/forex_rates",
+    "type": "drupal-drush",
+    "extra": {
+        "drush": {
+            "services": {
+                "drush.services.yml": "^10"
+            }
+        }
+    }
+}
```
Comment (joshirohit100) [inline docroot/modules/custom/forex_rates/composer.json]: we don't require this.

Comment (amansrivastava) [reply docroot/modules/custom/forex_rates/composer.json]: I was exploring ,module packaging and how it works with Composer.
It is also scaffolded from drush command. I can remove if not required.

**Comment by amansrivastava** [comment]
having conflicts with PR. closing this and will create a new one.

**Pull request closed without merging**